mlir/test/Conversion/RockToGPU/misc_ops.mlir

```diff
@@ -30,3 +31,22 @@ module {
     return
   }
 }
+
+// -----
+
+// CHECK: #[[MAP:.+]] = affine_map<(d0)[s0] -> (-d0 + s0 - 1)> 
+module {
+  func.func @misckernel(%arg0: memref<?xf32>, %arg1: memref<?xf32>)
+      attributes {kernel = 0 : i32, block_size = 64 : i32, grid_size = 900 : i32, reverse_grid} {
+    // CHECK: %[[BLOCKID:.+]] = gpu.block_id x
+    %bid = rock.workgroup_id : index
+    // CHECK: %[[REV_BLOCKID:.+]] = affine.apply #[[MAP]](%[[BLOCKID]])[%c900]
```

Comment:
I (personally) found this clearer to see in the IR that is being reversed rather than a substraction.

Reply:
But by the time we've lowered to `gpu` we're already in a low-level state where we have where we've even got `cf` instead of `scf` - that is, things like affine apply lowering have already happened.

Reply:
> affine apply lowering 

not really, see https://github.com/ROCm/rocMLIR/blob/develop/mlir/lib/Dialect/Rock/Pipelines/Pipelines.cpp#L208-L210